chirp/drivers/kenwood_live.py

```diff
@@ -198,39 +246,72 @@ def __init__(self, *args, **kwargs):
+        if isinstance(number, str):
+            return cmd, "%i,0,%s" % (self._vfo, number)
+        return cmd, "%i,0,%03i" % (self._vfo, number)
 
     def _cmd_get_memory_name(self, number):
         return "MNA", "%i,%03i" % (self._vfo, number)
 
-    def _cmd_get_split(self, number):
-        return "MR", "%i,1,%03i" % (self._vfo, number)
+    def _cmd_get_split(self, number, cmd="MR"):
+        if isinstance(number, str):
+            return cmd, "%i,1,%s" % (self._vfo, number)
+        return cmd, "%i,1,%03i" % (self._vfo, number)
 
-    def _cmd_set_memory(self, number, spec):
+    def _cmd_set_memory(self, number, spec, cmd="MW"):
         if spec:
             spec = "," + spec
-        return "MW", "%i,0,%03i%s" % (self._vfo, number, spec)
+        if isinstance(number, str):
+            return cmd, "%i,0,%s%s" % (self._vfo, number, spec)
+        return cmd, "%i,0,%03i%s" % (self._vfo, number, spec)
 
     def _cmd_set_memory_name(self, number, name):
         return "MNA", "%i,%03i,%s" % (self._vfo, number, name)
 
-    def _cmd_set_split(self, number, spec):
-        return "MW", "%i,1,%03i,%s" % (self._vfo, number, spec)
+    def _cmd_set_split(self, number, spec, cmd="MW"):
+        return cmd, "%i,1,%03i,%s" % (self._vfo, number, spec)
 
     def get_raw_memory(self, number):
         return command(self.pipe, *self._cmd_get_memory(number))
 
     def get_memory(self, number):
-        if number < 0 or number > self._upper:
+        name = None
+        cmdnum = number
+        if self._program_scan_lower is not None \
+                and self._program_scan_upper is not None \
+                and number >= self._program_scan_lower \
+                and number <= self._program_scan_upper:
+            read_cmd = "MR"
+            cmdnum -= self._program_scan_lower
+            if (number % 2) == (self._program_scan_lower % 2):
+                cmdnum = "L%d" % int(cmdnum / 2)
+            else:
+                cmdnum = "U%d" % int(cmdnum / 2)
+            name = "PScan %s" % cmdnum
+        elif self._callmem_lower is not None \
+                and self._callmem_upper is not None \
+                and number >= self._callmem_lower \
+                and number <= self._callmem_upper:
+            read_cmd = "CR"
+            cmdnum -= self._callmem_lower
+            name = "Call %d" % (cmdnum)
+        elif number < 0 or number > self._upper:
             raise errors.InvalidMemoryLocation(
                 "Number must be between 0 and %i" % self._upper)
+        else:
+            read_cmd = "MR"
         if number in self._memcache and not NOCACHE:
             return self._memcache[number]
 
-        result = command(self.pipe, *self._cmd_get_memory(number))
+        result = command(self.pipe, *self._cmd_get_memory(cmdnum, read_cmd))
         if result == "N" or result == "E":
             mem = chirp_common.Memory()
+            if self._has_name:
+                if name is not None:
```

Comment:
You're using this wrong. The `name` property is owned by the user and is only the alpha tag. If you can't have an alpha tag on these then you should set it to the empty string and make it immutable. What you're looking for is "special channels". You need to expose `valid_special_chans` in your `RadioFeatures`. You still need to give them a virtual `mem.number` like you are here, but set the name (like "Call 1") as `mem.extd_number`. That tells the UI that it's special and not part of the regular number memories. Check out some of the other drivers in the tree that expose those, and maybe open up an image file (in `tests/images`) of one to get an idea.

Reply:
Thanks, will do... it was very hard to get it working this way, so I suspect I was doing this wrong. :D